packages/preact/src/index.ts

```diff
@@ -80,9 +80,20 @@ function Text(this: AugmentedComponent, { data }: { data: Signal }) {
 			}
 		}
 
+		const defaultUpdaterCallback = this._updater!._callback;
+
 		// Replace this component's vdom updater with a direct text one:
 		this._updater!._callback = () => {
```

Comment:
We could also stop changing the `_callback` when we see that the signal isn't a primitive value. In that aspect the name of this function might be misleading as we might need to just call it `SignalValue` or something and opt in to these optimizations selectively

Reply:
Ok will rename it. So I think it should be renamed in react adapter too

Reply:
Renaming isn't the only course of action here I feel, we can also just conditionally override the callback rather than adding a check in the altered callback